adapters/gamma/gamma.go

```diff
@@ -197,21 +239,33 @@ func (a *GammaAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalReq
 		}}
 	}
 
-	var bidResp openrtb.BidResponse
-	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
+	var gammaResp gammaBidResponse
+	if err := json.Unmarshal(response.Body, &gammaResp); err != nil {
 		return nil, []error{&errortypes.BadServerResponse{
 			Message: fmt.Sprintf("bad server response: %d. ", err),
 		}}
 	}
 
-	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(bidResp.SeatBid[0].Bid))
-	for _, sb := range bidResp.SeatBid {
+	//(Section 7.1 No-Bid Signaling)
+	if gammaResp.SeatBid == nil || len(gammaResp.SeatBid) == 0 {
+		return nil, nil
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(gammaResp.SeatBid[0].Bid))
+	for _, sb := range gammaResp.SeatBid {
 		for i := range sb.Bid {
-			bid := sb.Bid[i]
-			bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
-				Bid:     &bid,
-				BidType: getMediaTypeForImp(bidResp.ID, internalRequest.Imp),
-			})
+			mediaType := getMediaTypeForImp(gammaResp.ID, internalRequest.Imp)
+			bid := convertBid(sb.Bid[i], mediaType)
+			if bid != nil {
+				bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+					Bid:     bid,
+					BidType: mediaType,
+				})
+			} else {
+				return nil, []error{&errortypes.BadServerResponse{
```

Comment:
Do you really want to discard all bids by returning nil here if any of them are missing their ad markup, or should you just discard the bid that's missing its ad markup but still return any other bids that were ok here?

Reply:
Oh, I almost forget it, thanks for notifying me.